grapple/types/pages.py

```diff
@@ -169,6 +169,66 @@ class Meta:
         interfaces = (PageInterface,)
 
 
+def get_preview_page(token):
+    """
+    Get a preview page from a token.
+    """
+    token_params_list = token.split(":").pop(0)
+    token_params_kvstr = token_params_list.split(";")
+
+    params = {}
+    for arg in token_params_kvstr:
+        key, value = arg.split("=")
+        params[key] = value
+
+    """
+    Is there a more convenient way to get at the get_page_from_preview_token
+    method? 
+    """
+
+    """ 
+    We get id when previewing a page that has been saved. We're going to use it
+    to get the page so we can lookup it's class in order to call 
+    get_page_from_preview_token
+    """
+    id = params.get("id")
+    if id:
+        page = WagtailPage.objects.get(pk=id).specific
+        if page:
+            cls = type(page)
+            """ 
+            is there a particular version of wagtail that requires checking that
+            get_page_from_preview_token exists? is there a version of wagtail 
+            at which we can drop this conditional safely?
```

Comment:
`get_page_from_preview_token` comes for wagtail-headless-preview and is Wagtail-versiona agnostic

Reply:
So this is really just a check to see if headless preview is installed and enabled?

Reply:
would we be better served with an isHeadlessPreviewEnabled() function?